src/colab_gitlab/signals.py

```diff
@@ -0,0 +1,105 @@
+
+import logging
+
+import requests
+
+from django.conf import settings
+from django.dispatch import receiver
+
+from colab.accounts.signals import user_password_changed, user_created
+
+LOGGER = logging.getLogger('colab.plugins.gitlab')
+
+
+@receiver(user_password_changed)
+def update_gitlab_password(sender, **kwargs):
+    user = kwargs.get('user')
+    password = kwargs.get('password')
+
+    app_config = settings.COLAB_APPS.get('colab_gitlab', {})
+    private_token = app_config.get('private_token')
+    upstream = app_config.get('upstream', '').rstrip('/')
+
+    error_msg = u'Error trying to update "%s" password on Gitlab. Reason: %s'
+
+    users_endpoint = '{}/api/v3/users'.format(upstream)
+
+    params = {'search': user.username, 'private_token': private_token}
+    try:
+        response = requests.get(users_endpoint, params=params)
+    except Exception as excpt:
+        reason = 'Request to API failed ({})'.format(excpt)
+        LOGGER.error(error_msg, user.username, reason)
+        return
+
+    users = response.json()
+
+    for gitlab_user in users:
+        if gitlab_user.get('username') == user.username:
+            break
+    else:
+        reason = u'Username not found on Gitlab'
+        LOGGER.error(error_msg, user.username, reason)
+        return
+
+    params = {'private_token': private_token, 'password': password}
+    request_url = '{}/{}'.format(users_endpoint, gitlab_user['id'])
+    try:
+        response = requests.put(request_url, params=params)
```

Comment:
This will be addressed in the next PR. You can take a look in `validate-passwd` branch.